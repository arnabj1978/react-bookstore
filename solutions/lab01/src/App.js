import './App.css';

function App() {
  return (
    <div className="App">
      <header>
        <h1>Welcome to React Bookstore</h1>
      </header>
      <main>
        <p>
          Here you can find the best books in the world. Click the Add to Cart
          button to add a book to your shopping cart.
        </p>
      </main>
    </div>
  );
}

export default App;
