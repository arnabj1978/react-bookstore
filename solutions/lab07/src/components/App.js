import Header from './Header';
import Main from './Main';
import Footer from './Footer';
import './App.css';
import productsData from '../data/products';

function App(props) {
  return (
    <div className="container">
      <Header />
      <Main products={productsData} />
      <Footer />
    </div>
  );
}

export default App;
