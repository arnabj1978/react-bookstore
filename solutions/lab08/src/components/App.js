import {useState} from 'react';
import Header from './Header'
import Main from './Main'
import Footer from './Footer';
import './App.css';
import productsData from '../data/products';

function App(props) {
  const [itemsInCart,setItemsInCart] = useState(['1','2','3']);

  return (
    <div className="container">
        <Header />
        <Main products = {productsData}
              itemsInCart = {itemsInCart} 
              setItemsInCart = {setItemsInCart} />
        <Footer />
    </div>
  );
}

export default App;
