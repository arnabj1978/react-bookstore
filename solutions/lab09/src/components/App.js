import { Component } from 'react';
import Header from './Header';
import Main from './Main';
import Footer from './Footer';
import './App.css';
import productsData from '../data/products';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      itemsInCart: [],
    };

    this.addToCart = this.addToCart.bind(this);
    this.removeFromCart = this.removeFromCart.bind(this);
  }

  addToCart(id) {
    let newItems = [...this.state.itemsInCart, id];
    this.setState({
      itemsInCart: newItems,
    });
  }

  removeFromCart(idToRemove) {
    let newItems = this.state.itemsInCart.filter((id) => id !== idToRemove);
    this.setState({ itemsInCart: newItems });
  }

  render() {
    return (
      <div className="container">
        <Header />

        <Main
          products={productsData}
          itemsInCart={this.state.itemsInCart}
          addToCart={this.addToCart}
          removeFromCart={this.removeFromCart}
        />
        <Footer />
      </div>
    );
  }
}

export default App;
