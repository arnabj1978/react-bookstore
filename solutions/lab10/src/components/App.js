import { Component } from 'react';
import Header from './Header';
import Main from './Main';
import Footer from './Footer';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      itemsInCart: [],
      products: [],
      loading: false,
    };
    this.addToCart = this.addToCart.bind(this);
    this.removeFromCart = this.removeFromCart.bind(this);
  }

  componentDidMount() {
    this.setState({ loading: true });
    fetch('http://localhost:3000/data/products.json').then((response) =>
      response
        .json()
        .then((products) => this.shuffleArray(products))
        .then((products) => {
          this.setState({ products: products, loading: false });
        })
    );
  }

  shuffleArray(array) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  addToCart(id) {
    let newItems = [...this.state.itemsInCart, id];
    this.setState({
      itemsInCart: newItems,
    });
  }

  removeFromCart(idToRemove) {
    let newItems = this.state.itemsInCart.filter((id) => id !== idToRemove);
    this.setState({ itemsInCart: newItems });
  }

  render() {
    return (
      <div className="container">
        <Header />
        <Main
          products={this.state.products}
          itemsInCart={this.state.itemsInCart}
          addToCart={this.addToCart}
          removeFromCart={this.removeFromCart}
        />
        <Footer />
      </div>
    );
  }
}

export default App;
