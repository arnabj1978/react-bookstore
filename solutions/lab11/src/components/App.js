import { useState, useEffect } from 'react';
import Header from './Header';
import Main from './Main';
import Footer from './Footer';
import './App.css';

function App() {
  const [itemsInCart, setItemsInCart] = useState([]);
  const [products, setProducts] = useState([]);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    async function fetchData() {
      try {
        setIsLoading(true);
        const response = await fetch('/data/products.json');
        const json = await response.json();
        shuffleArray(json);
        setProducts(json);
        setIsLoading(false);
      } catch (e) {
        console.error(e);
      }
    }
    fetchData();
  }, []);

  function shuffleArray(array) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  function addToCart(id) {
    let newItems = [...itemsInCart, id];
    setItemsInCart(newItems);
  }

  function removeFromCart(idToRemove) {
    let newItems = itemsInCart.filter((id) => id !== idToRemove);
    setItemsInCart(newItems);
  }

  if (isLoading) {
    return <p>Loading...</p>;
  }
  return (
    <div className="container">
      <Header />
      <Main
        products={products}
        itemsInCart={itemsInCart}
        addToCart={addToCart}
        removeFromCart={removeFromCart}
      />
      <Footer />
    </div>
  );
}

export default App;
