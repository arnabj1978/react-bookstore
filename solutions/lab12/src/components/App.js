import { useState, useEffect } from 'react';
import Header from './Header';
import Main from './Main';
import Footer from './Footer';
import './App.css';

function App() {
  const [itemsInCart, setItemsInCart] = useState([]);
  const [products, setProducts] = useState([]);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    async function fetchData() {
      try {
        setIsLoading(true);
        const response = await fetch(
          'http://localhost:3000/data/products.json'
        );
        const json = await response.json();
        setProducts(shuffleArray(json));
        setIsLoading(false);
      } catch (e) {
        console.error(e);
      }
    }
    fetchData();
  }, [setProducts]);

  function shuffleArray(array) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  function addToCart(id) {
    let newItems = [...itemsInCart, id];
    setItemsInCart(newItems);
  }

  function removeFromCart(idToRemove) {
    let newItems = itemsInCart.filter((id) => id !== idToRemove);
    setItemsInCart(newItems);
  }

  return (
    <div className="container">
      <Header />
      {isLoading ? 'Loading' : ''}
      <Main
        products={products}
        itemsInCart={itemsInCart}
        addToCart={addToCart}
        removeFromCart={removeFromCart}
      />
      <Footer />
    </div>
  );
}

export default App;
