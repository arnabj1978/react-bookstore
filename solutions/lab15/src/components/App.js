import { useState, useEffect } from 'react';
import Header from './Header';
import Main from './Main';
import Footer from './Footer';
import './App.css';
import * as actionCreators from '../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

function App(props) {
  const [isLoading, setIsLoading] = useState(false);
  const [shuffledProductList, setShuffledProductList] = useState([]);
  const { loadProducts } = props;

  useEffect(() => {
    async function fetchData() {
      try {
        setIsLoading(true);

        const response = await fetch(
          'http://localhost:3000/data/products.json'
        );
        let json = await response.json();
        json = await shuffleArray(json);
        loadProducts(json);

        setIsLoading(false);
      } catch (e) {
        console.error(e);
      }
    }
    fetchData();
  }, [loadProducts]);

  useEffect(() => {
    let shuffled = shuffleArray(props.products);
    setShuffledProductList(shuffled);
  }, [props.products]);

  function shuffleArray(array) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  return (
    <div className="container">
      <Header />
      {isLoading ? 'Loading' : ''}
      <Main
        products={shuffledProductList}
        itemsInCart={props.itemsInCart}
        addToCart={props.addToCart}
        removeFromCart={props.removeFromCart}
      />
      <Footer />
    </div>
  );
}
const mapStateToProps = (state, props) => {
  return {
    itemsInCart: state.cart.items,
    products: state.products.products,
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actionCreators, dispatch);
};

const wrappedApp = connect(mapStateToProps, mapDispatchToProps)(App);

export default wrappedApp;
