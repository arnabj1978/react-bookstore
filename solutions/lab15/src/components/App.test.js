import { render, screen } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  it('renders testing text', () => {
    render(<App />);
    const testText = screen.getByText(/Welcome to React Bookstore/i);
    expect(testText).toBeInTheDocument();
  });
  it('renders testing text', () => {
    render(<App />);
    const testText = screen.getByText(/Welcome to React Bookstore/i);
    expect(testText).toBeInTheDocument();
  });
  it('renders as expected', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
