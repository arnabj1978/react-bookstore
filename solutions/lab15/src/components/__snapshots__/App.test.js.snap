// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders as expected 1`] = `
<div>
  <div
    class="container"
  >
    <header>
      <h1>
        Welcome to React Bookstore
      </h1>
    </header>
    Loading
    <main
      class="row"
    >
      <div
        class="col-md-8"
      >
        <ul
          class="productList"
          data-testid="product-list"
        />
      </div>
      <div
        class="col-md-4"
      >
        <div
          class="cart"
        >
          <h2>
            Cart
          </h2>
          Total: $
          0
           USD
        </div>
      </div>
    </main>
    <footer>
      <p>
        This is the footer.
      </p>
    </footer>
  </div>
</div>
`;
